### PR TITLE
added the gui

### PR DESCRIPTION
**Pull request by Aliza596**
image

### CODE REVIEW

File: src/main/java/weizberg/gameoflife/GridComponent.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+                } else {
+                    grid.makeAlive(gridX, gridY);
+                }
+                for (int y = 0; y < getWidth(); y++) {
+                    for (int x = 0; x < getHeight(); x++) {
+                        if (grid.isAlive(x, y)) {
+                            System.out.println(x + ", " + y + " is alive");
+                        }
+                    }
+                }
```
Comment (andrewoid) [inline src/main/java/weizberg/gameoflife/GridComponent.java]: don't do this every mouse click

File: src/main/java/weizberg/gameoflife/GridComponent.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        int rows = field.length;
+        int cols = field[0].length;
+        for (int y = 0; y < rows; y++) {
+            for (int x = 0; x < cols; x++) {
+                if (grid.isAlive(x, y)) {
+                    g.fillRect(x * 20, y * 20, 20, 20);
+                }
+            }
+        }
+        repaint();
```
Comment (andrewoid) [inline src/main/java/weizberg/gameoflife/GridComponent.java]: If you're using a timer then you don't need repaint here, put it in the timer.

Comment (Aliza596) [reply src/main/java/weizberg/gameoflife/GridComponent.java:92]: When I moved repaint() to the timer, it does not paint when I click on a cell.

File: src/main/java/weizberg/gameoflife/GridComponent.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.util.Arrays;
+
+public class GridComponent extends JComponent {
+    private final Grid grid;
+    private int[][] field;
+    Timer timer = new Timer(1000, new ActionListener() {
```
Comment (andrewoid) [inline src/main/java/weizberg/gameoflife/GridComponent.java]: private

**Review by andrewoid — APPROVED** [review]

**Pull request merged**